agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,53 +106,16 @@ public CallReplacementTransformer() {}
       byte[] classfileBuffer)
       throws IllegalClassFormatException {
 
-    // debug = className.equals ("chicory/Test");
-
+    debug_transform.log("In transform: class = %s%n", className);
     String fullClassName = className.replace("/", ".");
-
-    debug_transform.log("In Transform: class = %s%n", className);
-
-    // Don't instrument boot classes. We only want to instrument
-    // user classes classpath.
-    // Most boot classes have the null loader,
-    // but some generated classes (such as those in sun.reflect) will
-    // have a non-null loader. Some of these have a null parent loader,
-    // but some do not. The check for the sun.reflect package is a hack
-    // to catch all of these. A more consistent mechanism to determine
-    // boot classes would be preferrable.
-    if (loader == null) {
-      debug_transform.log("ignoring system class %s, class loader == null", fullClassName);
-      return (null);
-    } else if (loader.getParent() == null) {
-      debug_transform.log("ignoring system class %s, parent loader == null\n", fullClassName);
-      return (null);
-    } else if (fullClassName.startsWith("sun.reflect")) {
-      debug_transform.log("ignoring system class %s, in sun.reflect package", fullClassName);
-      return (null);
-    } else if (fullClassName.startsWith("com.sun")) {
-      System.out.printf("Class from com.sun package %s with nonnull loaders\n", fullClassName);
-    }
-
-    // Don't intrument our code
-    if (className.startsWith("randoop.")) {
-      debug_transform.log("Not considering randoop class %s%n", fullClassName);
-      return (null);
-    }
-
-    // Look for match with specified regular expressions for class
-    Map<MethodDef, ReplacementClass> current_method_map = null;
-    debug_class = false;
-    for (MethodMapInfo mmi : map_calls_file) {
-      if (mmi.class_regex.matcher(className).matches()) {
-        if (false && className.startsWith("RandoopTest")) debug_class = true;
-        if (debug_class)
-          System.out.printf("Classname %s matches re %s%n", className, mmi.class_regex);
-        current_method_map = mmi.map;
-        break;
+    if (isBootClass(loader, fullClassName)) {
```

Comment:
If nothing else, the reason for this would be to explicitly exclude classes that you aren't going to transform.  However, using a simple test case, skipping these classes seems to only save about 40 seconds wall-clock in Gradle.